spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -56,24 +66,58 @@ private void setSortOrders(SortOrder[] sortOrdersInfo) {
   }
 
   public static Transform[] toGravitinoPartitionings(
-      org.apache.spark.sql.connector.expressions.Transform[] transforms) {
+      org.apache.spark.sql.connector.expressions.Transform[] transforms, boolean isHiveProvider) {
     if (ArrayUtils.isEmpty(transforms)) {
       return Transforms.EMPTY_TRANSFORM;
     }
 
-    return Arrays.stream(transforms)
-        .filter(transform -> !isBucketTransform(transform))
-        .map(
-            transform -> {
-              if (transform instanceof IdentityTransform) {
-                IdentityTransform identityTransform = (IdentityTransform) transform;
-                return Transforms.identity(identityTransform.reference().fieldNames());
-              } else {
-                throw new NotSupportedException(
-                    "Doesn't support Spark transform: " + transform.name());
-              }
-            })
-        .toArray(Transform[]::new);
+    List<Transform> gravitinoTransforms =
```

Comment:
Is `BucketTransform` a distribution for Hive, a partition for Iceberg?  I would like to add a interface or variable like `supportsBucketPartition` (or other better name) to control the transform logic not `isHiveProvider`, because `supportsXX` is more general to support other catalogs.

Reply:
`supportsXX` will change the logic `isBucketTransform`, then all transform logic could be unified,  WDYT?

Reply:
it looks more reasonable, thanks.

Reply:
> Is `BucketTransform` a distribution for Hive, a partition for Iceberg?
yes.

Reply:
updated.